rpc-rs/src/otel.rs

```diff
@@ -0,0 +1,243 @@
+//! Contains helpers and code for enabling [OpenTelemetry](https://opentelemetry.io/) tracing for
+//! wasmbus-rpc calls. Please note that right now this is only supported for providers. This module
+//! is only available with the `otel` feature enabled
+
+use std::collections::HashSet;
+
+/// NOTE: The commented out code in here will work once we upgrade to async_nats. I have left it so that we don't have to rewrite it again
+// use async_nats::header::HeaderName;
+// use async_nats::HeaderMap;
+use nats_aflowt::header::HeaderMap;
+use opentelemetry::propagation::{Extractor, Injector, TextMapPropagator};
+use opentelemetry::sdk::propagation::TraceContextPropagator;
+use tracing::span::Span;
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+lazy_static::lazy_static! {
+    static ref EMPTY_HEADERS: HeaderMap = HeaderMap::default();
+}
+
+// /// A convenience type that wraps a NATS [`HeaderMap`] and implements the [`Extractor`] trait
```

Comment:
If this large block of commented out stuff is not going to be uncommented, we should probably remove it.

Reply:
Woops, I didn't see it while sitting through on git hub. Carry on, nothing to see here.